### PR TITLE
handle rejection promises gracefully

### LINKED ISSUE #14: unhandled promise rejections
We store a lot of promises in an Array without await'ing them directly. If any of those rejects, this is considered an unhandlded promise rejection and might cause the node process to terminate.

### PR DESCRIPTION
**Pull request by Germandrummer92**
closes #14.

* ran npm audit fix
* one should _never_ have unawaited promises as they will lead to UnhandledPromiseRejections if they reject in the time UNTIL they will be awaited. This fixes that, see test case.

### CODE REVIEW

File: src/handledRejectionPromise.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  constructor(promise: Promise<T>) {
+    this.storedPromise = promise.then(
+      (value) => {
+        this.value = { type: "resolved", value };
+        return;
+      },
+      (reason: unknown) => {
+        this.value = { type: "rejected", reason };
+      }
+    );
```
Comment (marc-at-work) [inline src/handledRejectionPromise.ts:28]: I don't see a reason for not using async await syntax here, same below for the promise getter

**Comment by kosta** [comment]
> your description: "if they reject in the time UNTIL they will be rejected" should probably be "if they reject in the time UNTIL they will be handled"? :D

Yes! I think @Germandrummer92 fixed it already.

File: src/handledRejectionPromise.ts (hunk 1 above)
Comment (marc-at-work) [reply src/handledRejectionPromise.ts:28]: Yeah, that's what I had in mind, but I understand that one can find it "iffy" :D

will merge it tomorrow or on monday or so

Comment (kosta) [reply src/handledRejectionPromise.ts:28]: :D

Shall I replace the `get promise()` with `async promise()`?

Comment (marc-at-work) [reply src/handledRejectionPromise.ts:28]: no, it's fine :)

**Pull request merged**